### PR TITLE
First pass at input action refactor

### PR DESCRIPTION
**Pull request by StephenHodgson**
Preliminary work on Input Action Profile refactor.

I'll try to list as many of the changes as possible going into this. 

- Simplified the controller implementations a bit by providing a `InteractionMapping` wrapper for raising all the events to the input system.
- Removed all the rule profiles and data structures.
- Stubbed in the input action profile, but there's still a lot that needs to go in there.

### CODE REVIEW

File: XRTK-Core/Packages/com.xrtk.core/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "Mixed Reality",
     "DI"
   ],
-  "version": "0.1.20",
+  "version": "0.1.21",
```
Comment (SimonDarksideJ) [inline XRTK-Core/Packages/com.xrtk.core/package.json]: A little premature updating this for a feature branch :D

Comment (StephenHodgson) [reply XRTK-Core/Packages/com.xrtk.core/package.json]: Nope. Always increment for breaking changes.

Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/package.json]: Doctors also used to always just saw legs off at the first sign of infection.   Never say "Always"

**Pull request closed without merging**